docs/publish/deploy-office-add-in-sso-to-azure.md

```diff
@@ -0,0 +1,253 @@
+    
+    /* global console, process, require, __dirname */
+    
+    const app = express();
+    const port = process.env.PORT;
+    
+    app.set("port", port);
+    
+    // view engine setup
+    app.set("views", path.join(__dirname, "views"));
+    app.set("view engine", "pug");
+    
+    app.use(logger("dev"));
+    app.use(express.json());
+    app.use(express.urlencoded({ extended: false }));
+    app.use(cookieParser());
+    
+    /* Turn off caching when developing */
+    if (process.env.NODE_ENV !== "production") {
+      app.use(express.static(path.join(process.cwd(), "dist"), { etag: false }));
+    
+      app.use(function (req, res, next) {
+        res.header("Cache-Control", "private, no-cache, no-store, must-revalidate");
+        res.header("Expires", "-1");
+        res.header("Pragma", "no-cache");
+        next();
+      });
+    } else {
+      // In production mode, let static files be cached.
+      app.use(express.static(path.join(process.cwd())));
+      console.log("static set up: " + path.join(process.cwd()));
+    }
+    
+    const indexRouter = express.Router();
+    indexRouter.get("/", function (req, res) {
+      //   res.render("/taskpane.html");
+      res.sendFile("/taskpane.html", { root: __dirname });
+    });
+    
+    app.use("/", indexRouter);
+    
+    app.get("/getuserdata", validateJwt, getUserData);
+    
+    // Catch 404 and forward to error handler
+    app.use(function (req, res, next) {
+      next(createError(404));
+    });
+    
+    // error handler
+    app.use(function (err, req, temp, res) {
+      // set locals, only providing error in development
+    
+      res.locals.message = err.message;
+      res.locals.error = req.app.get("env") === "development" ? err : {};
+    
+      // render the error page
+      res.status(err.status || 500).send({
+        message: err.message,
+      });
+    });
+    
+    app.listen(process.env.PORT, () => console.log("Server listening on port: " + process.env.PORT));
+    ```
+
+1. Save the file.
+
+## Update app registration
+
+We recommend you create multiple app registrations for localhost, staging, and deployment testing. The following steps ensure that the app registration you use for deployment correctly uses the app service URL.
+
+1. In the Azure portal, open your app registration. Note that the app registration may be in a different account than your app service. Be sure to sign in to the correct account.
+1. In the left sidebar, select **Authentication**.
+1. On the **Authentication** pane, find the `https://localhost:3000/fallbackauthdialog.html` and change it to use the app service URL you saved previously. For example, `https://contoso.sso.azurewebsites.net/fallbackauthdialog.html`.
+1. Save the change.
+1. In the left sidebar, select **Expose an API**.
+1. Change the **Application ID URI** to use your app service URL you saved previously. For example, `api://contoso-sso.azurewebsites.net/628050c7-8d46-4f8f-a393-ac22eb688477`.
```

Comment:
I got this error here:
![image](https://user-images.githubusercontent.com/38896772/209881877-584d092a-b16b-48c5-9f05-2dc8a3e47652.png)

Reply:
It should be `api:` protocol, not https. I'll see if I can clear that up with the copy paste instructions.